extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -166,16 +166,25 @@ export class Bind {
     /** @type {!Array<./bind-evaluator.BindingDef>} */
     const bindings = [];
 
-    const elements = this.elementsInNode_(body);
-    const numElements = elements.length;
+    const doc = dev().assert(node.ownerDocument, 'ownerDocument is null.');
+    const walker = doc.createTreeWalker(node, NodeFilter.SHOW_ELEMENT);
 
-    // Helper function for scanning a slice of elements.
+    /**
+     * Helper function for scanning a slice of elements.
+     * Returns true if there are no more elements to scan.
+     * @param {number} start
+     * @param {number} end
+     * @return {boolean}
+     */
     const scanFromTo = (start, end) => {
```

Comment:
Thanks for the suggestion. Simplified the helper method and extracted the loop. 

I kept the advancement of the tree walker in the helper to avoid duplicating the conditional with `break` above. I also left it as a closure because [modifying input params is bad for code readability](http://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern).